pcawg_metadata_parser/generate_pcawg_lane_level_gnos_analysis_ids.py

```diff
@@ -0,0 +1,243 @@
+    if isinstance(obj, set):
+        return list(obj)
+    raise TypeError
+
+def get_formal_repo_name(repo):
+    repo_url_to_repo = {
+      "https://gtrepo-bsc.annailabs.com/": "bsc",
+      "bsc": "https://gtrepo-bsc.annailabs.com/",
+      "https://gtrepo-ebi.annailabs.com/": "ebi",
+      "ebi": "https://gtrepo-ebi.annailabs.com/",
+      "https://cghub.ucsc.edu/": "cghub",
+      "cghub": "https://cghub.ucsc.edu/",
+      "https://gtrepo-dkfz.annailabs.com/": "dkfz",
+      "dkfz": "https://gtrepo-dkfz.annailabs.com/",
+      "https://gtrepo-riken.annailabs.com/": "riken",
+      "riken": "https://gtrepo-riken.annailabs.com/",
+      "https://gtrepo-osdc-icgc.annailabs.com/": "osdc-icgc",
+      "osdc-icgc": "https://gtrepo-osdc-icgc.annailabs.com/",
+      "https://gtrepo-osdc-tcga.annailabs.com/": "osdc-tcga",
+      "osdc-tcga": "https://gtrepo-osdc-tcga.annailabs.com/",
+      "https://gtrepo-etri.annailabs.com/": "etri",
+      "etri": "https://gtrepo-etri.annailabs.com/"
+    }
+
+    return repo_url_to_repo.get(repo)
+
+def init_report_dir(metadata_dir, report_name):
+    report_dir = metadata_dir + '/reports/' + report_name
+    if os.path.exists(report_dir): shutil.rmtree(report_dir, ignore_errors=True)  # empty the folder if exists
+    os.makedirs(report_dir)
+
+    return report_dir
+
+
+def main(argv=None):
+
+    parser = ArgumentParser(description="PCAWG Full List of GNOS entities Info Generator",
+             formatter_class=RawDescriptionHelpFormatter)
+    parser.add_argument("-m", "--metadata_dir", dest="metadata_dir",
+             help="Directory containing metadata manifest files", required=True)
+
+    args = parser.parse_args()
+    metadata_dir = args.metadata_dir  # this dir contains gnos manifest files, will also host all reports
+
+    if not os.path.isdir(metadata_dir):  # TODO: should add more directory name check to make sure it's right
+        sys.exit('Error: specified metadata directory does not exist!')
+
+    timestamp = str.split(metadata_dir, '/')[-1]
+    es_index = 'p_' + re.sub(r'\D', '', timestamp).replace('20','',1)
+    es_type = "donor"
+    es_host = 'localhost:9200'
+
+    es = Elasticsearch([es_host])
+
+    # output result
+    report_name = re.sub(r'^generate_', '', os.path.basename(__file__))
+    report_name = re.sub(r'\.py$', '', report_name)
+    report_dir = init_report_dir(metadata_dir, report_name)
+
+
+	# get the list of donors in PCAWG
+    donors_list = get_donors_list(es, es_index, es_queries)
+    
+    report_info_list_full = []
+    # get json doc for each donor and reorganize it 
+    for donor_unique_id in donors_list:            
+    	es_json = get_donor_json(es, es_index, donor_unique_id)
+        report_info_list_donor = create_gnos_entity_info(donor_unique_id, es_json)
+        report_info_list_full.extend(report_info_list_donor)
+
+                
+    for repo in ['bsc', 'ebi', 'cghub', 'dkfz', 'riken', 'osdc-icgc', 'osdc-tcga', 'etri']:
```

Comment:
We probably need to remove cghub from this list. CGHub's input lane-level BAMs are normal sequence submission from TCGA projects to CGHub.